lib/src/license_detection/license.dart

```diff
@@ -131,6 +174,38 @@ List<License> licensesFromFile(String path) {
 /// Regex to match the all the text starting from `END OF TERMS AND CONDTIONS`.
 const _endOfTerms = 'END OF TERMS AND CONDITIONS';
 
+/// Generates crc-32 checksum for the given list of tokens
+/// by taking [granularity] token values at a time.
+List<Ngram> generateChecksums(List<Token> tokens, int granularity) {
+  if (tokens.length < granularity) {
+    final text = tokens.join(' ');
+    return [Ngram(text, crc32(utf8.encode(text)), 0, tokens.length - 1)];
+  }
+
+  var nGrams = <Ngram>[];
+
+  for (var i = 0; i + granularity <= tokens.length; i++) {
+    var text = '';
+    tokens.sublist(i, i + granularity).forEach((token) {
```

Comment:
consider using `List.skip(i).take(granularity)` instead of `List.subList` especially if you're just going to iterate through the values... `List.subList` will always create a new list object. `skip/take` will just create an iterator for the range.

Reply:
I wanted to use `take` but I was not aware of the `skip` function, there are lot of instances of using `sublist`. I'll change them.